### DISCUSSION TITLE
BitKeep Wallet Connector

### DISCUSSION THREAD

**Discussion by chenjiji** [wagmi-dev/references#35] [Connector request]
Hello, every one.

BitKeep is a one-stop transaction encryption wallet APP. [BitKeep Website](https://bitkeep.io/)
In order to better serve our wallet users, we need to create a BitKeep-specific connector. Of course, it is isolated and will not affect the normal use of other connectors.
We named it: BitKeepConnector, and it has been developed and tested. We can discuss some questions and suggestions about it later.

thank you very much.

**Comment by tmm** [comment]
Can you [go through the requirements](https://github.com/wagmi-dev/references/blob/main/.github/CONTRIBUTING.md#connectors) and post responses here.

**Reply by chenjiji** [reply]
I've read the docs carefully and I'm ready.

**Reply by tmm** [reply]
This Connector [does not solve for or provide a novel use-case](https://github.com/wagmi-dev/references/blob/main/.github/CONTRIBUTING.md#1-the-connector-must-present-a-novel-use-case), since it extends `InjectedConnector` without any significant differences. In this case, instead of a separate Connector, folks can use `InjectedConnector`'s [`getProvider`](https://wagmi.sh/react/connectors/injected#getprovider) if they want to target BitKeep.

**Reply by dellwatson** [reply]
ok but it doenst explained properly either. imnot sure how to use inject provider.

tried it several things like these (and none of them connecting).
```js
  new InjectedConnector({
      chains,
      options: {
        name: "BitKeep",
        getProvider: () =>
          typeof window !== "undefined" ? window?.bitkeep : undefined,
      },
    }),
    ```
    
    or
    ```js
  new InjectedConnector({
      chains,
      options: {
        name: "BitKeep",
        getProvider: () =>
          typeof window !== "undefined" ? window?.isBitkeep : undefined,
      },
    }),
    ```
    
     or
    ```js
  new InjectedConnector({
      chains,
      options: {
        name: "BitKeep",
        getProvider: () =>
          typeof window !== "undefined" ? window?.BitKeepJs : undefined,
      },
    }),
    ```
